cmd/create/operatorroles/common_utils.go

```diff
@@ -19,3 +24,30 @@ func computePolicyARN(accountID string, prefix string, namespace string, name st
 	}
 	return fmt.Sprintf("arn:%s:iam::%s:policy/%s", aws.GetPartition(), accountID, policy)
 }
+
+func validateIngressOperatorPolicyOverride(r *rosa.Runtime, policyArn string, sharedVpcRoleArn string,
+	installerRolePrefix string) error {
+	_, err := r.AWSClient.IsPolicyExists(policyArn)
+	policyExists := err == nil
+	if !policyExists {
+		return nil
+	}
+
+	policyDocument, err := r.AWSClient.GetDefaultPolicyDocument(policyArn)
+	if err != nil {
+		return err
+	}
+
+	// The policy associated with the installer role. In the case it contains a different shared VPC role ARN,
+	// don't override it.
+	if strings.Contains(policyDocument, assumePolicyAction) && !strings.Contains(policyDocument, sharedVpcRoleArn) {
```

Comment:
NP:
Would it be possible to retrieve the existing shared vpc arn listed in the op role assume policy?
To allow a more transparent error message like:

```
"Policy with ARN 'A' contains 'sts:AssumeRole' action with an unexpected shared VPC role ARN. [Expected: 'B', Provided: 'C']
To create operator roles with shared vpc role ARN 'C', please provide a different value for '--installer-role-arn'."
```

Reply:
I agree that the error message you're suggesting is more helpful.
We have a place in the code where we parse the policy document JSON, and then it would be possible, let me play with the code and see what it looks like.
I took this approach, of `strings.Contains`, because it was sort of a last-minute change, and tried to keep the PR as simple as possible.

Reply:
lets merge and get this through QE asap, we can follow up with error message improvements.

Reply:
Created this ticket https://issues.redhat.com/browse/OCM-3522 to follow up on the next iteration.
Thanks for the feedback.